tasks/model.go

```diff
@@ -24,6 +25,19 @@ type Task struct {
 	StorageTask   *StorageTask   `json:"storage_task,omitempty"`
 }
 
+func (t Task) Bytes() []byte {
+	b, err := json.Marshal(&t)
+	if err != nil {
+		return []byte{}
```

Comment:
This worries me a bit - if for some reason we get a json error, we'd then happily send an empty task, signing those empty bytes too.

We just have one call site, so I'd do the json.Marshal and double error check there. If we find that too tedious and repeated in a bunch of places, we could add a function or method on an entire AuthenticatedTask, like Sign or Verify.

Reply:
Here's another detail to consider - json.Marshal isn't canonical/stable, and it's certainly possible that the encoding would change over time e.g. if encoding/json gets bugfixes or if we change the struct type.

Can this possibly affect us? E.g. if we signed the task when it was assigned, upgrade the controller while it's in progress, and later end up with different json-encoded bytes when we want to verify a signature.

If you think that might be an issue, one possible solution would be to store the originally signed bytes next to the signature. Then we're also free to change the encoding entirely, for example.

Reply:
eventually we probably want this over the data in ipld schema (once we get to reputation system). I suspect this code changes a couple more times as we move in that direction